### DISCUSSION TITLE
Flat arrows

### DISCUSSION THREAD

**Discussion by jhurliman** [foxglove/community#1078] [General]
**Description**

With ROS arrow markers, you can set `size` to `{1, 1, 0.01}` to get a flat arrow. There doesn't seem to be any way to replicate this with SceneEntity arrows.

- Version: latest
- Platform: all

**Steps To Reproduce**
* Fail to make a flat SceneEntity arrow

**Expected Behavior**
* Succeed in making a flat SceneEntity arrow

**Comment by defunctzombie** [comment]
Are you not able to set `head_length` to 0? https://docs.foxglove.dev/docs/visualization/message-schemas/arrow-primitive/

What is a "flat arrow"? Like only the head?

**Comment by jhurliman** [comment]
Unfortunately, no combination of those two values will produce an arrow that looks like this (ignore the bevels):

![archicad_arrow_object](https://github.com/foxglove/schemas/assets/195374/a100675e-44d0-4601-ab91-05a66f672332)

**Comment by defunctzombie** [comment]
Is that a screenshot of the arrow you get from the ROS arrow marker or the ArrowPrimitive?

**Comment by jhurliman** [comment]
It's a random image I found online that shows a flat arrow, in the same style that the ROS arrow marker can produce. This type of style cannot be replicated with an `ArrowPrimitive` because you can only control head/shaft length and diameter, you can't resize per 3d axis.

**Comment by defunctzombie** [comment]
> It's a random image I found online that shows a flat arrow, in the same style that the ROS arrow marker can produce. This type of style cannot be replicated with an `ArrowPrimitive` because you can only control head/shaft length and diameter, you can't resize per 3d axis.

Gotcha. Thanks for the additional context. I'm gonna move this to feature cause our current arrow primitive is designed for making (volumetric?) arrows rather than trying to be a 1:1 copy of the ROS Marker message.

**Comment by jhurliman** [comment]
Drawing flat arrows in 3D space is a useful viz tool. Also useful for projecting into 2D camera view
